### PR TITLE
Add workarounds in pyHIPIFY for __forceinline__ and std:: math functions in HIP

### PR DESCRIPTION
**Pull request by jithunnair-amd**

### CODE REVIEW

**Comment by petrex** [comment]
@rohithkrn

File: tools/amd_build/pyHIPIFY/hipify-python.py [hunk 1]
```diff
@@ -331,6 +331,18 @@ def disable_asserts(input_string):
         output_string = output_string.replace(input_string[start:p_end + 1], "")
     return output_string
 
+def replace_forceinline(input_string):
```
Comment (Jorghi12) [inline tools/amd_build/pyHIPIFY/hipify-python.py]: Why is this function needed?

Comment (iotamudelta) [reply tools/amd_build/pyHIPIFY/hipify-python.py]: @jithunnair-amd could you add a doc string here with the link to HIP not implying static for forceinline?

**Comment by iotamudelta** [comment]
network error once more
@pytorchbot  retest this please

**Comment by iotamudelta** [comment]
Merging this - even though I am not seeing these issues at compile them (nor does the CI), we have multiple independent confirmations of them. The changes are relatively innocent: changing forceinline to inline makes sense as by the documented HIP shortcoming. Removing the std:: for some math functions fixes problems and doesn't regress anything.

**Comment by iotamudelta** [comment]
apparently I got an outdated state for checks failed.

@pytorchbot retest this please

**Comment by iotamudelta** [comment]
@jithunnair-amd it seems that it complains about missing the yaml module. Can you add it through pip?

**Comment by Jorghi12** [comment]
@iotamudelta 
This is where the dependencies are being installed. It's using pip which installs to **python2** by default.
https://github.com/ROCmSoftwarePlatform/pytorch/blob/master/.jenkins/pytorch/build.sh#L30

File: tools/amd_build/pyHIPIFY/hipify-python.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
```
Comment (Jorghi12) [inline tools/amd_build/pyHIPIFY/hipify-python.py]: This should be reverted if you want to avoid the yaml missing import error.

**Comment by iotamudelta** [comment]
@jithunnair-amd could you also add a note to the function from our offline discussion conc. expf/exp? Thanks!

**Comment by iotamudelta** [comment]
Finally gotten all checks passed - merging this now w/ the caveat that @jithunnair-amd and I are going to talk to hcc group about std::* math functions.

File: tools/amd_build/pyHIPIFY/hipify-python.py (hunk 1 above)
Comment (AlexVlx) [reply tools/amd_build/pyHIPIFY/hipify-python.py]: @iotamudelta please don't do this, the correct fix is to change how HIP handles `__forceinline__` to match what CUDA does (it actually implies language `inline` there). Please ask the HIP maintainers to correct that.

Comment (iotamudelta) [reply tools/amd_build/pyHIPIFY/hipify-python.py]: @AlexVlx yes, we will. for the time being we need things to compile reliably in all our environments. it's a hack, unfortunately. @jithunnair-amd could you create HIP issue w/ that and link it here for documentation purposes? thanks! :-)

Comment (jithunnair-amd) [reply tools/amd_build/pyHIPIFY/hipify-python.py]: https://github.com/ROCm-Developer-Tools/HIP/issues/572

**Pull request merged**